### PR TITLE
Add support for comment blocks to sysctl resource

### PR DESCRIPTION
**Pull request by lanky**
### Description
Adds support for comments in sysctl files under sysctl.d - permits sysctl settings to be documented/explained for troubleshooting (and informational) purposes.

The comment property is optional and accepts both Strings and Arrays (of strings). 

### Issues Resolved
N/A
### Check List

- [ ] New functionality includes tests
- [X] All tests pass
- [X ] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)
- [ X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [X ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: lib/chef/resource/sysctl.rb [hunk 1]
```diff
@@ -45,6 +45,10 @@ class Sysctl < Chef::Resource
                coerce: proc { |v| coerce_value(v) },
                required: true
 
+      property :comment, [Array, String],
+               description: "An optional comment to prepend the resource setting.",
```
Comment (tas50) [inline lib/chef/resource/sysctl.rb]: We generate our docs from these comments so you'll want to briefly explain that multi-line comments can be accomplished by using an array of strings where each line is a value in the array.

Comment (lanky) [reply lib/chef/resource/sysctl.rb]: OK, have updated and pushed.  Thanks.

Comment (lanky) [reply lib/chef/resource/sysctl.rb]: hmmm, that didn't work. All tests pass locally. moving it back to a single line to test.

File: lib/chef/resource/sysctl.rb [hunk 2]
```diff
@@ -45,6 +45,10 @@ class Sysctl < Chef::Resource
                coerce: proc { |v| coerce_value(v) },
                required: true
 
+      property :comment, [Array, String],
+               description: "Optional comments, placed above the resource setting in the generated file. For multi-line comments, use an array of strings, one per line.",
```
Comment (lanky) [inline lib/chef/resource/sysctl.rb]: the only change in the last commit is to make this description longer. Not sure why the tests are now failing?

**Pull request closed without merging**